File: client/src/app/_models/process/candidateFlight.ts


export interface ICandidateFlight
{
    id: number;
    depId: number;
    depItemId: number;
    cvRefId: number;
    applicationNo: number;
    candidateName: string;
    customerName: string;
    customerCity: string;

    airlineName: string;
    flightNo: string;
    airportOfBoarding: string;
    airportOfDestination: string;
    eTD_Boarding: Date;
    eTA_Destination: Date;
    airportVia: string;
    flightNoVia: string;
    eTA_Via?: Date | null;
    eTD_Via?: Date | null;
    fullPath: string;

    fileToUpload: File|null;

}

export class CandidateFlight implements ICandidateFlight
{
    id=0;
    depId = 0;
    depItemId=0;
    cvRefId=0;
    applicationNo=0;
    candidateName='';
    customerName='';
    customerCity='';

    airlineName = '';
    flightNo = ''; 
    airportOfBoarding = '';
    flightNoVia = ''; 
    airportOfDestination = '';
    eTD_Boarding= new Date;
    eTA_Destination= new Date;
    airportVia = '';
    eTD_Via= null;    //new Date;
    eTA_Via= null;  // new Date;
    fullPath='';
    fileToUpload= null;
}

File: api/Entities/Admin/CandidateFlight.cs
using System.ComponentModel.DataAnnotations;
using api.Entities.Deployments;

namespace api.Entities.Admin
{
    public class CandidateFlight: BaseEntity
    {
        public int DepId { get; set; }
        public int DepItemId { get; set; }
        public int CvRefId { get; set; }
        public int ApplicationNo { get; set; }
        public string CandidateName { get; set; }
        public string CustomerName { get; set; }
        public string CustomerCity { get; set; }
        public DateTime DateOfFlight { get; set; }
        
        public string FlightNo { get; set; }
        public string AirportOfBoarding { get; set; }
        public string AirportOfDestination { get; set; }
        public DateTime ETD_Boarding { get; set; }
        public DateTime ETA_Destination { get; set; }
        public string AirportVia { get; set; }
        public string FightNoVia { get; set; }
        public DateTime? ETA_Via { get; set; }
        public DateTime? ETD_Via { get; set; }
        [MaxLength(250)]
        public string FullPath {get; set;}
    }
}